tools/grpc_generator/generate_stubs.py

```diff
@@ -0,0 +1,59 @@
+"""Script to generate gRPC stubs for all packages in the repository."""
+
+import json
+import os
+import subprocess
+
+
+def _get_path_from_root_directory_relative_path(root_directory_relative_path) -> str:
+    return os.path.join(
+        os.path.dirname(__file__),
+        "..",
+        "..",
+        _convert_path_to_use_os_path_separators(root_directory_relative_path),
+    )
+
+
+def _convert_path_to_use_os_path_separators(path) -> str:
+    return path.replace("/", os.path.sep)
+
+
+packages_file_name = "packages.json"
+packages_file_path = _get_path_from_root_directory_relative_path(packages_file_name)
+
+with open(packages_file_path, "r", encoding="utf-8") as packages_file:
+    packages = json.load(packages_file)
+
```

Comment:
Best practice: write a main() function, otherwise your helper functions can see the locals assigned below.